src/ServiceControl.Monitoring/Http/Diagrams/TimingsAggregator.cs

```diff
@@ -39,6 +44,23 @@ public IEnumerable<MonitoredEndpoint> AggregateIntoLogicalEndpoints()
                     CriticalTime = AggregateTimings(instanceCriticalTime)
                 };
 
+                var queueLength = queueLengths.FirstOrDefault(ql => ql.Key == endpointName);
+                var queueLengthValues = queueLength.Value?.OrderBy(kvp => kvp.Key).ToArray();
+
+                if (queueLengthValues != null)
+                {
+                    var queueLengthMinDate = queueLength.Value.Min(kvp => kvp.Key);
+
+                    monitoredEndpoint.QueueLength = new LinearMonitoredValues
+                    {
+                        Average = queueLength.Value?.Values.Average(),
```

Comment:
Haven't we established that queueLength.Value is non-null by the time we get here?